recipe/migrations/r-base42.yaml

```diff
@@ -0,0 +1,10 @@
+migrator_ts: 1661453785
+__migrator:
+  kind: version
+  migration_number: 1
+  bump_number: 1
```

Comment:
Why is the recipe number bumped for the migration to R 4.2? This causes a new R 4.1 version to be uploaded as well, which seems unnecessary, at least to me. It wouldn't really matter for a few recipes, but this is going to upload thousands of essentially duplicate r41 binaries to anaconda.org (the only difference being the build number).

Reply:
Even if the build number was not bumped, there'll be a new version uploaded because some of the pinnings (like compiler) have changed and there's a different hash than the one when it was last rebuilt. Therefore bumping the build number avoids two packages with R 4.1 with the same build number, but different hashes.